File: docs/codes/templates/datarobot_mlops_package-7.2.1_python/examples/python/SampleCode/regression_sample_code.py
import time
import random
import pandas as pd

from datarobot.mlops.mlops import MLOps

"""
This sample code is intended to demonstrate how the mlops library is called in the documentation.
It does not have real data (or even a real model) and should not be run against a real MLOps
service.
"""


def _make_fake_feature_df(num_samples):

    f1 = [random.random() for i in range(num_samples)]
    f2 = [random.random() for i in range(num_samples)]
    f3 = [random.random() for i in range(num_samples)]

    # create a dictionary of feature name to array of feature values
    feature_data = {
        "feature1": f1,
        "feature2": f2,
        "feature3": f3
    }

    return pd.DataFrame.from_dict(feature_data)


def _make_fake_sample_predictions_list(num_samples):
    return [random.random() for i in range(num_samples)]


def main():
    num_samples = 10

    # MLOPS: initialize mlops library
    # If deployment ID is not set, it will be read from MLOPS_DEPLOYMENT_ID environment variable.
    # If model ID is not set, it will be ready from MLOPS_MODEL_ID environment variable.
    mlops = MLOps().init()

    features_df = _make_fake_feature_df(num_samples)

    start_time = time.time()
    predictions_array = _make_fake_sample_predictions_list(num_samples)
    end_time = time.time()

    # MLOPS: report the number of predictions in the request and the execution time.
    mlops.report_deployment_stats(len(predictions_array), (end_time - start_time) * 1000)

    # MLOPS: report the prediction results.
    mlops.report_predictions_data(features_df=features_df, predictions=predictions_array)

    # MLOPS: release MLOps resources when finished.
    mlops.shutdown()


if __name__ == "__main__":
    main()

File: docs/codes/templates/datarobot_mlops_package-7.2.1_python/examples/java/SampleCode/src/main/java/com/datarobot/mlops/examples/RegressionSampleCode.java

package com.datarobot.mlops.examples;

import com.datarobot.mlops.MLOps;
import com.datarobot.mlops.common.exceptions.DRCommonException;
import com.datarobot.mlops.common.enums.SpoolerType;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Random;

public class RegressionSampleCode {
    static private Random rand;
    private String deploymentId;
    private String modelId;

    RegressionSampleCode(String deploymentId, String modelId) {
        this.deploymentId = deploymentId;
        this.modelId = modelId;
        rand = new Random();
    }

    Map<String, List<Object>> makeFakeFeatureData(int numSamples) {

        Map<String, List<Object>> featureData = new HashMap<>();

        List<Object> feature1Values = new ArrayList<>();
        // feature1 is a double
        for (int i = 0; i < numSamples; i++) {
            feature1Values.add(rand.nextDouble());
        }
        featureData.put("feature1_name", feature1Values);

        // feature2 is a string
        List<Object> feature2Values = new ArrayList<>();
        for (int i = 0; i < numSamples; i++) {
            feature2Values.add("some_text" + i);
        }
        featureData.put("feature2_name", feature2Values);

        return featureData;
    }

    public List<Double> makeFakeRegressionPredictionsList(int numSamples) {
        List<Double> predictions = new ArrayList<>(numSamples);
        for (int i = 0; i < numSamples; i++) {
            predictions.add(rand.nextDouble());
        }

        return predictions;
    }

    /**
     * Sample code demonstrating how to use MLOps library to record metrics on regression predictions.
     * All code specific to MLOps library are commented with "MLOPS"
     */
    public void run(int numSamples) {

        // MLOPS: declaration
        MLOps mlops = null;
        try {

            // MLOPS: initialize mlops instance.
            // If the deployment ID is not specified, the environment variable MLOPS_DEPLOYMENT_ID is used.
            // If the model ID is not specified, the environment variable MLOPS_MODEL_ID is used.
            // If the output type is not specified, the environment variable MLOPS_SPOOLER_TYPE is used.
            mlops = MLOps.getInstance()
                    .setDeploymentId(deploymentId)
                    .setModelId(modelId)
                    .setStdoutSpooler()
                    .init();

            // Create a set of feature values for each sample
            // For this example, we simply generate the feature values.
            Map<String, List<Object>> featureData = makeFakeFeatureData(numSamples);

            // Call the model to generate predictions.
            // For this example, we simply fake the generating a random number of predictions.
            long startTime = System.currentTimeMillis();
            List<Double> predictions = makeFakeRegressionPredictionsList(numSamples);
            long endTime = System.currentTimeMillis();

            // MLOPS: report the feature data and predictions together to mlops
            mlops.reportPredictionsData(featureData, predictions);

            // MLOPS: report number of samples and time to produce predictions
            mlops.reportDeploymentStats(numSamples, endTime - startTime);


        } catch (DRCommonException e) {
            System.out.println(e.getMessage());
        } finally {
            // MLOPS: shutdown
            if (mlops != null) {
                mlops.shutdown();
            }
        }
    }

    public static void main(String args[]) {
        int numSamples = 10;
        String deploymentId = "sample_deployment_id";
        String modelId = "sample_model_id";
        RegressionSampleCode regressionExample = new RegressionSampleCode(deploymentId, modelId);
        regressionExample.run(numSamples);
    }
}



